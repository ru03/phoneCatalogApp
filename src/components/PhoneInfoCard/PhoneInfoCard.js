import React from 'react';
import classes from './PhoneInfoCard.module.css';

const phoneInfoCard = (props) => (
    <article className={classes.Card}>
        <header className={classes.Header}>
            <img src="https://vignette.wikia.nocookie.net/battlefordreamisland/images/c/cd/IPhone_X.png/revision/latest?cb=20180625201840"></img>
            <h1>{props.name}</h1>
        </header>
        <section className={classes.Section}>
            <p>Model: {props.desc}</p>
            <button>Detalles</button>
        </section>
    </article>
)

export default phoneInfoCard;