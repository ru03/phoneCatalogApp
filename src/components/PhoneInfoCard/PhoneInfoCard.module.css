.Card {
    box-shadow: 0 2px 1px -1px rgba(0,0,0,.2), 
                0 1px 1px 0 rgba(0,0,0,.14),
                0 1px 3px 0 rgba(0,0,0,.12);
    max-width: 400px;
    background-color: white;
    color: black;
    margin: 10px;
}

.Header {
    display: flex;
    flex-flow: column;
    align-items: center;
}

.Section {
    display: flex;
    flex-direction: column;
    margin: 10px;
}

img {
    max-width: 90%;
}