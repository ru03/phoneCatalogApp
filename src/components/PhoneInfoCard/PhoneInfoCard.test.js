import React from 'react';
import { mount, shallow } from 'enzyme';
import PhoneInfoCard from './PhoneInfoCard';

describe('PhoneInfoCard component test', () => {
    let phoneInfo;

    beforeEach(() => {
        phoneInfo = shallow(<PhoneInfoCard></PhoneInfoCard>);
    });

    it('PhoneInfoCard should be displayed', () => {
        expect(phoneInfo).toMatchSnapshot();
    });

    it('PhoneInfoCard click event should be called', () => {
        const handlerClick = jest.fn();
        const phoneWithProps = mount(<PhoneInfoCard clicked={handlerClick}></PhoneInfoCard>)
        phoneWithProps.props().clicked()
        expect(handlerClick).toBeCalled();
    });
    
    it('PhoneInfoCard click event should return an id', () => {
        const handlerClick = jest.fn(id => 1)
        const phoneWithProps = mount(<PhoneInfoCard clicked={handlerClick}></PhoneInfoCard>)
        phoneWithProps.props().clicked()
        expect(handlerClick()).toEqual(1);
    });

    it('PhoneInfoCard props should be filled up', () => {
        const phoneWithProps = mount(<PhoneInfoCard image='image'
            model='iphone8'
            brand='apple'>
        </PhoneInfoCard>)
        expect(phoneWithProps.props().model).toEqual('iphone8');
        expect(phoneWithProps.props().brand).toEqual('apple');
        expect(phoneWithProps.props().image).toEqual('image');
    });
})