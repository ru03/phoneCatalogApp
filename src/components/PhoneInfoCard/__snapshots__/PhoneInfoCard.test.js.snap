// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInfoCard component test PhoneInfoCard should be displayed 1`] = `
<article
  className="Card"
>
  <header
    className="Header"
  >
    <img />
    <h1 />
  </header>
  <section
    className="Section"
  >
    <p>
      Brand: 
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Details
    </button>
  </section>
</article>
`;
